shared/base/router.js

```diff
@@ -166,7 +168,21 @@ _.extend(BaseRouter.prototype, Backbone.Events, {
             callback.apply(this, arguments);
           }
         }
-        realAction.call(this, params, next);
+        // in AMD environment realAction is the string containing path to the controller
+        // which will be loaded async (might be preloaded)
+        // Only used in AMD environment
+        if (isAMDEnvironment) {
+          requireAMD([realAction], function(controller) {
```

Comment:
Then let's check for the variable type, which will protect us from changes down the road
and will allow more flexible treatment of AMD environment on the server.